src/helpers/transport/bytearrstream/mod.rs

```diff
@@ -0,0 +1,62 @@
+mod aligned;
+
+pub use aligned::ByteArrStream as AlignedByteArrStream;
+
+use crate::error::BoxError;
+use axum::extract::BodyStream;
+use futures::Stream;
+use futures_util::{stream, TryStreamExt};
+use hyper::body::Bytes;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+/// represents the item of an underlying stream
+type Item = Result<Bytes, BoxError>;
+/// represents the underlying stream
+type PinnedStream = Pin<Box<dyn Stream<Item = Item> + Send>>;
+
+pub struct ByteArrStream {
+    stream: PinnedStream,
+}
+
+impl ByteArrStream {
+    #[must_use]
+    pub fn new(stream: PinnedStream) -> Self {
+        Self { stream }
+    }
+
+    #[must_use]
+    #[allow(clippy::missing_panics_doc)] // `size_in_bytes` known to be small
+    pub fn align(self, size_in_bytes: usize) -> AlignedByteArrStream {
+        AlignedByteArrStream::new(self.stream, u32::try_from(size_in_bytes).unwrap())
+    }
+}
+
+impl From<BodyStream> for ByteArrStream {
+    fn from(stream: BodyStream) -> Self {
+        ByteArrStream::new(Box::pin(stream.map_err(<BoxError>::from)) as PinnedStream)
+    }
+}
+
+impl From<Vec<u8>> for ByteArrStream {
```

Comment:
no need to change, just fyi that `From<Box<[u8]>>` is more generic as every vector pointer can be cast to `Box`

Reply:
good to know